### PR TITLE
[TASK] Get all nodes in a single request in ReferencesEditor

### PR DESCRIPTION
**Pull request by dfeyer**
This change introduce a new method in the NodeSearchServiceInterface
to find node by identifiers. The NodeController and the
ReferencesEditor is adapted to use this new method based on the
given request arguments.

Releases: master
Resolves: NEOS-1261

### CODE REVIEW

**Comment by dfeyer** [comment]
:+1:

**Comment by dfeyer** [comment]
Can be done in more clean way, if we improve the NodeSearchInterface, see:

https://discuss.neos.io/t/rfc-improve-nodesearchserviceinterface/591

**Comment by hlubek** [comment]
@dfeyer: Could you merge the current master to have Travis CI running on this PR?

I'd suggest to not change the `NodeSearchServiceInterface` and just implement the fetching of Nodes via the public (TYPO3CR) Context API in the controller to get this PR merge-ready. Once the search service would be capable of doing this, it's easy to refactor the controller to use the search service for the task.

**Comment by dfeyer** [comment]
@hlubek I just merged master on this PR, but look like Travis is like me ... tired ... see you tomorrow

**Comment by kitsunet** [comment]
The early PRs are somewhat broken in that regard, regardless of closing/opening. You might want to create a new PR from the branch.

**Comment by dfeyer** [comment]
Tested on recent master work fine.

@hlubek NodeSearchServiceInterface is not API currently. I agree that this can be done in more clean way. But with the UI refactoring we will build more solid API for searching. 

Currently this change make Neos usable if you have tons of references, the current situation is pretty frustrating for user (I even have one of those user on my right, who discover this PR with a lots of pleasure).

For me :+1: for merging this one. And continue to work on building a more solid API for node searching.

File: TYPO3.Neos/Classes/TYPO3/Neos/Controller/Service/NodesController.php [hunk 1]
```diff
@@ -98,7 +99,11 @@ public function indexAction($searchTerm = '', $workspaceName = 'live', array $di
         }
 
         $contentContext = $this->createContentContext($workspaceName, $dimensions);
-        $nodes = $this->nodeSearchService->findByProperties($searchTerm, $searchableNodeTypeNames, $contentContext, $contextNode);
+        if ($nodeIdentifiers === array()) {
+            $nodes = $this->nodeSearchService->findByProperties($searchTerm, $searchableNodeTypeNames, $contentContext, $contextNode);
+        } else {
+            $nodes = $this->nodeSearchService->findByIdentifiers($nodeIdentifiers, $contentContext, $contextNode);
```
Comment (hlubek) [inline TYPO3.Neos/Classes/TYPO3/Neos/Controller/Service/NodesController.php]: @dfeyer Using

    array_map(function($identifier) { return $contentContext->getNodeByIdentifier($identifier); }, $nodeIdentifiers)

would almost implement the new search service method and we don't need to refactor yet another method for the UI rewrite, so why implement a new search service method?

My issue with this is that every new code that uses the NodeDataRepository is a big smell and makes it harder to refactor later.

**Comment by dfeyer** [comment]
@hlubek Apply your suggestion, NodeSearchServiceInterface is back to the previous state

**Comment by mgoldbeck** [comment]
:+1:

**Comment by loop86** [comment]
:+1:

**Comment by kitsunet** [comment]
:+1:

**Pull request merged**